doc: memfault: update README (#12204)
nrfconnect/sdk-nrf: samples/debug/memfault/README.rst

Requirements
************

> Before this sample can be used make sure to have registered a user at `Memfault registration page`_ and `create a new project in Memfault`_.

Reviewer: ```suggestion
Before using this sample, make sure to sign up in the `Memfault registration page`_ and `create a new project in Memfault`_.
```

> Before this sample can be used make sure to have registered a user at `Memfault registration page`_ and `create a new project in Memfault`_.
> You will be directed to the Integration guide for Memfault.
> You will receive the ``project key`` here to set in the :kconfig:option:`CONFIG_MEMFAULT_NCS_PROJECT_KEY` Kconfig option.

Reviewer: ```suggestion
Before using this sample, make sure to complete the steps of Memfault integration into |NCS|. 
See :ref:`ug_memfault` for details.
```
These steps are mentioned in the Memfault integration guide : https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/external_comp/memfault.html#prerequisites . So, we could link the user guide here. We could remove line no:13 as we link the same user guide there also.

Also, we could move the suggested text after the sample yaml table (Step no:24), considering the user flow.

Author: Yes, I could agree to that, but at the same time the memfault integration guide takes you through a few more steps with the configuration files, which is already integrated in the sample, so I feel like that at the same time would be a bit misleading. It is really just the prerequisite in the memfault integration guide you need to have set in order for the sample to work.

Before testing, ensure that your device is configured with the project key of your Memfault project.
|test_sample|

> 

Reviewer: ```suggestion
```
The extra line could be removed.

Before testing, ensure that your device is configured with the project key of your Memfault project.
|test_sample|

1.  |connect_terminal_ANSI|

> There is also possible to connect with the kit using the nrf Connect for VS Code extension and `connect to a serial nrf Serial Terminal. <How to connect to the terminal>`

Reviewer: We have NCSDK-20386 Jira open to change the recommended serial terminal to VSC terminal in all samples and applications. May be remove this for now as this be consider in this Jira.
